source/objects/bodyClass.m

```diff
@@ -168,6 +171,47 @@ function checkinputs(obj)
                 warning(['yaw using (default) 1 deg interpolation threshold.' newline 'Ensure this is appropriate for your geometry'])
             end
         end
+
+        function checkHydroInputs(obj)
+            % This method checks WEC-Sim user inputs for each hydro body and generates error messages if parameters are not properly defined for the bodyClass.
+            
+            % Check Morison Element Inputs for option 1
+            if obj.morisonElement.option == 1
+                [rgME,~] = size(obj.morisonElement.rgME);
+                [rz,~] = size(obj.morisonElement.z);
+                if rgME > rz
+                    obj.morisonElement.z = NaN(rgME,3);
+                end
+                clear rgME rz
+            end
+            % Check Morison Element Inputs for option 2
+            if obj.morisonElement.option == 2
+                [r,~] = size(obj.morisonElement.z);
+                for ii = 1:r
+                    if norm(obj.morisonElement.z(ii,:)) ~= 1
+                        error(['Ensure the Morison Element .z variable is a unit vector for the ',num2str(ii),' index'])
+                    end
+                end
+            end
+            % Warning for cg and cb being overwritten
+            if ~isempty(obj.cg) || ~isempty(obj.cb)
+                warning('Center of gravity and center of buoyancy are overwritten by h5 data for hydro bodies.')
+            end
+        end
+        
+        function checkDragNonHydroInputs(obj,ii)
+            % This method checks WEC-Sim user inputs for each non-hydro
+            % body and generates error messages if parameters are not properly defined for the bodyClass.
+            if isempty(obj.cg)
```

Comment:
Is there a reason to split these up? Could we just keep them in `body.checkInputs()`, and pass the `nonHydroBodLogic` logic, etc? Asking for consistency with the other classes that only have `checkInputs()`